src/libraries/Common/src/System/Security/Cryptography/Asn1/AsnXml.targets

```
@@ -21,23 +19,25 @@
       XmlInputPaths="@(AsnXml)"
       OutputPaths="@(AsnXml -> '$(IntermediateOutputPath)asnxml\%(filename).cs')" />
 
-    <Exec Condition="'$(TargetOS)'=='windows'"
+    <Exec Condition="$([MSBuild]::IsOsPlatform('windows')) == 'true'"
       IgnoreExitCode="true"
       StandardOutputImportance="Low"
       Command="$(SystemRoot)\System32\fc.exe /a @(AsnXml -> '$(IntermediateOutputPath)asnxml\%(filename).cs') @(AsnXml -> '%(Identity).cs')">
       <Output TaskParameter="ExitCode" ItemName="_AsnXmlDiffCode" />
     </Exec>
 
-    <!-- TODO: Call diff on Unix -->
-    <ItemGroup Condition="'$(TargetOS)'!='windows'">
-      <_AsnXmlDiffCode Include="1" />
-    </ItemGroup>
+    <Exec Condition="$([MSBuild]::IsOsPlatform('windows')) != 'true'"
+      IgnoreExitCode="true"
+      StandardOutputImportance="Low"
+      Command="/usr/bin/diff -q -a @(AsnXml -> '$(IntermediateOutputPath)asnxml\%(filename).cs') @(AsnXml -> '%(Identity).cs')">
```

I thiiiiink `/usr/bin/diff` on all non-Windows platforms? It is on Ubuntu Linux and macOS. It felt a little better to me to specify a full path instead.

---

I think it would be portable if we add hardcoded path as a fallback to `command(1)`:

```suggestion
      Command="%24%28command -v diff || echo /usr/bin/diff%29 -q -a @(AsnXml -> '$(IntermediateOutputPath)asnxml\%(filename).cs') @(AsnXml -> '%(Identity).cs')">
```

---

Hmm, isn't `command` a dash/bash builtin? Are there any concerns around this not being portable?

---

`command -v diff` is POSIX shelll way of finding the location. https://pubs.opengroup.org/onlinepubs/009604499/utilities/command.html
Depending on package manager / distro / OS, we can have `diff(1)` installed in `/usr/local/bin` or under `/opt` etc.

---

Gotcha. Thank you!